In [3]:
import numpy as np
from scipy import stats # importando scipy.stats
import pandas as pd # importando pandas
import matplotlib.pyplot as plt # importando matplotlib
import seaborn as sns
from scipy.stats import skew
import scipy.stats as st

# Pair Programming Estadística III

Vamos ahora a tomar de nuevo el conjunto de datos con el que 
estuvimos trabajando en los últimos días y vamos a continuar 
con el análisis de las estadísitcas de los contenidos del conjunto de datos.

Para todas los ejercicios de pair programming de
Estadística I, II y III usaremos el dataframe que 
se tome para el desarrollo de este ejercicio de pair programming.

Se ruega a la hora de realizar la entrega que incluyais el conjunto de datos que hayais decidido emplear para estos ejercicios.

# Objetivos
Los objetivos de hoy son:
 Extraer los siguentes estadísiticos del conjunto de datos para varias de las varibles disponibles:

### Tabla de contingencia

In [4]:
df_csv = pd.read_csv("01-imdb_top_1000.csv" )
df_csv.head(2)

Poster_Link  \
0  https://m.media-amazon.com/images/M/MV5BMDFkYT...   
1  https://m.media-amazon.com/images/M/MV5BM2MyNj...   

               Series_Title Released_Year Certificate  Runtime         Genre  \
0  The Shawshank Redemption          1994           A  142 min         Drama   
1             The Godfather          1972           A  175 min  Crime, Drama   

   IMDB_Rating                                           Overview  Meta_score  \
0          9.3  Two imprisoned men bond over a number of years...        80.0   
1          9.2  An organized crime dynasty's aging patriarch t...       100.0   

               Director          Star1           Star2       Star3  \
0        Frank Darabont    Tim Robbins  Morgan Freeman  Bob Gunton   
1  Francis Ford Coppola  Marlon Brando       Al Pacino  James Caan   

            Star4  No_of_Votes        Gross  
0  William Sadler      2343110   28,341,469  
1    Diane Keaton      1620367  134,966,411

In [16]:
df_csv.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 18 columns):
 #   Column         Non-Null Count  Dtype   
---  ------         --------------  -----   
 0   Poster_Link    1000 non-null   object  
 1   Series_Title   1000 non-null   object  
 2   Released_Year  1000 non-null   object  
 3   Certificate    899 non-null    object  
 4   Runtime        1000 non-null   object  
 5   Genre          1000 non-null   object  
 6   IMDB_Rating    1000 non-null   float64 
 7   Overview       1000 non-null   object  
 8   Meta_score     843 non-null    float64 
 9   Director       1000 non-null   object  
 10  Star1          1000 non-null   object  
 11  Star2          1000 non-null   object  
 12  Star3          1000 non-null   object  
 13  Star4          1000 non-null   object  
 14  No_of_Votes    1000 non-null   int64   
 15  Gross          831 non-null    object  
 16  votos          1000 non-null   category
 17  meta_score     843 non-null    cat

In [5]:
# cuantitativas
votos = df_csv['No_of_Votes']
meta = df_csv['Meta_score']
imdb = df_csv['IMDB_Rating']
anio = df_csv['Released_Year']

In [12]:
#cualitativas
genero = df_csv['Genre']
certificado = df_csv['Certificate']
director = df_csv['Director']

In [32]:
votos_crosstab = pd.crosstab(certificado,genero)
print(votos_crosstab)

Genre        Action, Adventure  Action, Adventure, Comedy  \
Certificate                                                 
16                           0                          0   
A                            2                          1   
Approved                     0                          0   
G                            0                          0   
GP                           0                          0   
PG                           0                          0   
PG-13                        0                          0   
Passed                       0                          1   
R                            0                          2   
TV-14                        0                          0   
TV-MA                        0                          0   
TV-PG                        0                          0   
U                            1                          2   
U/A                          0                          0   
UA                      

### Coeficiente de correlación de las variables numéricas del dataset.

In [28]:
df_cor = np.corrcoef(meta, imdb)
df_cor

array([[nan, nan],
       [nan,  1.]])

### Sesgos de alguna de las variables numéricas del dataset.

In [29]:
sesgo_votos = skew(votos)
print(sesgo_votos)

2.2965591310736615


### Intervalo de confianza de algunas de las variables numéricas del dataset.

In [30]:
print(st.t.interval(alpha=0.99, df=len(votos)-1, loc=np.mean(votos), scale=st.sem(votos)))

print("El intervalo de confianza de votos al 99% es", np.mean(votos))

print(st.t.interval(alpha=0.99, df=len(imdb)-1, loc=np.mean(imdb), scale=st.sem(imdb)))

print("El intervalo de confianza de imdb al 99% es", np.mean(imdb))


(246975.76748985777, 300410.0545101423)
El intervalo de confianza de votos al 99% es 273692.911
(7.9268169461659586, 7.97178305383404)
El intervalo de confianza de imdb al 99% es 7.949299999999999


### Interpretación de los resultados.

Ahora interpreta los resultados obtenidos de los diferentes estadísticos estudiados haciendo hincapié en aspectos interesantes de tus datos y explica lo mejor posible la naturaleza de los mismos.

Datos obtenidos:

- Tabla de contingencia: Es un tabla imposible hemso probado con todas las variables y salen muchisimas siempre, esta a sido la mas pequeña y tiene 191 columnas. Mirando la tabla podemos decir que :
La mayoria de las peliculas de genero Sci-Fi tiene un certificado  UA.

Los thriller obtienen diferentes certificados no suelen ser siempre los mismos suelen ser el A y el UA. 

Para los Western  lso certificados suelen ser A o U.

- Coeficiente de correlación:

array([[nan, nan],
       [nan,  1.]])

Aquí la correlación no existe.

- Sesgo

2.2965591310736615

Al ser un sesgo positivo, nuetros datos estarían sesgados hacia la derecha, pero lo que realmente ocurre es que tenemos más películas de mayor metascore (las valoraciones son en decenas frente a IMDB que son unidades), no es que esté sesgada nuestra muestra, sino que es así.

- Intervalos de confianza:

    · Votos = 273692.911

       (246975.76748985777, 300410.0545101423)

    · IMDB = 7.949299999999

       (7.9268169461659586, 7.97178305383404)

Hemos elegido unos intervalos de confianza del 99% que quieren decir que según el número de votos lo más seguro es que estuviera entre 2046975 y 300410, ya que todos nuestros datos tienen números elevados.

Para el IMDB, lo más seguro es si le otorgan una calificación a una nueva película que introdujéramos, lo más seguro es que estaría entre esos valores.